security/storage.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2016: Mirantis Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import collections
+import datetime
+import logging
+
+from security import elastic
+
+
+class Storage:
+
+    def __init__(self, config):
+        logging.debug("Loading backend with config %s", config)
+        self.backend = elastic.Backend(**config)
+
+    def update_issues(self, region, issues, issue_types):
+        """This method should be called by checker.
+
+        :param region:
+        :param issues: generator
+        :param issue_types: List of issue types supported by plugin
+        """
+        now = datetime.datetime.utcnow()
+        self.backend.begin()
+        issues = list(issues)
+
+        #  Create mapping of discovered issues
+        discovered_issues = collections.defaultdict(dict)
+        for issue in issues:
+            discovered_issues[issue.type][issue.id] = issue
+
+        #  Get all issues from backend
+        stored_issues = collections.defaultdict(dict)
+        for issue in self.backend.get_issues(region, issue_types):
```

Comment:
If the issues are never deleted from the backend — this would blow up with time. we can probably leave this as issue and fix later (i.e. find a way to query elastic for issues with `resolved_at` present in the documents)